cmd/crates/stellar-ledger/src/lib.rs

```diff
@@ -74,69 +73,60 @@ pub enum LedgerError {
     Bip32PathError(String),
 
     #[error(transparent)]
-    XdrError(#[from] XdrError),
+    XdrError(#[from] xdr::Error),
 
     #[error(transparent)]
     DecodeError(#[from] DecodeError),
 }
 
-pub struct LedgerOptions<T: Exchange> {
-    exchange: T,
-    hd_path: slip10::BIP32Path,
-}
-
 pub struct LedgerSigner<T: Exchange> {
-    network_passphrase: String,
     transport: T,
-    hd_path: slip10::BIP32Path,
 }
 
 impl<T> LedgerSigner<T>
 where
     T: Exchange,
 {
+    pub fn new(transport: T) -> Self {
+        Self { transport }
+    }
+    pub fn native() -> Result<LedgerSigner<TransportNativeHID>, Error> {
+        Ok(LedgerSigner {
+            transport: get_transport()?,
+        })
+    }
+}
+
+impl<T> Blob for LedgerSigner<T>
+where
+    T: Exchange,
+{
+    type Key = HdPath;
+    type Error = Error;
     /// Get the public key from the device
     /// # Errors
     /// Returns an error if there is an issue with connecting with the device or getting the public key from the device
-    pub async fn get_public_key(
+    async fn get_public_key(
```

Comment:
Are we able to remove the `pub` here because it is part of implementing a public trait, and so it implicitly also public?

Reply:
Yeah I'm surprised my ide didn't yell at me about this.